src/Signal.elm

```diff
@@ -148,7 +149,9 @@ update wins, just like with `merge`.
 -}
 mergeMany : List (Signal a) -> Signal a
 mergeMany signals =
-    List.foldr1 merge signals
+    case signals of
+        s :: rest -> List.foldr merge s rest
```

Comment:
I did a reverse and foldl. I'm pretty sure this is equivalent to foldr, but please check if over again.